libraries/AP_MotionController/AP_MotionController.cpp

```diff
@@ -0,0 +1,758 @@
+    // @Range: 0 1e6
+    // @Units: DN
+    // @Increment: 1
+    // @User: Advanced
+    // <<<<<<<<<<AP_GROUPINFO("PDB4", 55, AP_MotionController, rc[3].m2.pos.deadband, AP_MOTIONCONTROLLER_POS_DEADBAND),
+
+    // @Param: PMX4
+    // @DisplayName: SERVO_MOCR_PMX4
+    // @Description: Position controller maximum position limit for fourth roboclaw
+    // @Range: 0 1e9
+    // @Units: QP
+    // @Increment: 1
+    // @User: Advanced
+    // <<<<<<<<<<AP_GROUPINFO("PMX4", 56, AP_MotionController, rc[3].m2.pos.maxpos, AP_MOTIONCONTROLLER_POS_MAXPOS),
+
+    // @Param: PMN4
+    // @DisplayName: SERVO_MOCR_PMN4
+    // @Description: Position controller minimum position limit for fourth roboclaw
+    // @Range: 0 1e9
+    // @Units: QP
+    // @Increment: 1
+    // @User: Advanced
+    // <<<<<<<<<<AP_GROUPINFO("PMN4", 57, AP_MotionController, rc[3].m2.pos.minpos, AP_MOTIONCONTROLLER_POS_MINPOS),
+
+    // @Param: PAC4
+    // @DisplayName: SERVO_MOCR_PAC4
+    // @Description: Position controller acceleration for fourth roboclaw
+    // @Range: 0 1e9
+    // @Units: QPPS/S
+    // @Increment: 1
+    // @User: Advanced
+    AP_GROUPINFO("PAC4", 58, AP_MotionController, rc[3].m2.acceleration, AP_MOTIONCONTROLLER_POS_ACCEL),
+
+    // @Param: PVE4
+    // @DisplayName: SERVO_MOCR_PVE4
+    // @Description: Position controller velocity for fourth roboclaw
+    // @Range: 0 1e9
+    // @Units: QPPS
+    // @Increment: 1
+    // @User: Advanced
+    AP_GROUPINFO("PVE4", 59, AP_MotionController, rc[3].m2.velocity, AP_MOTIONCONTROLLER_POS_VEL),
+
+    AP_GROUPEND};
+
+AP_MotionController::AP_MotionController()
+{
+    // set defaults from the parameter table
+    AP_Param::setup_object_defaults(this, var_info);
+}
+
+AP_MotionController::~AP_MotionController()
+{
+}
+
+void AP_MotionController::init(void)
+{
+    AP_SerialManager &serial_manager = AP::serialmanager();
+    port = serial_manager.find_serial(AP_SerialManager::SerialProtocol_ESCTelemetry, 0);
+    roboclaw.init(port, 10000);
+
+    // Check if the serial port communication successed
+    char buff[48];
+    bool bSuccess = false;
+    for (int i = 0; i < 4; i++)
+    {
+        if (roboclaw.ReadVersion(rc[i].address.get(), buff))
+        {
+            hal.console->printf("RoboClaw[%d] Version: %s\n", i, buff);
+            bSuccess = true;
+        }
+        else
+        {
+            hal.console->printf("RoboClaw[%d] is not connected or responding\n", i);
+        }
+    }
+    // Reset roboclaw communication port
+    if (bSuccess == false)
+    {
+        roboclaw.init(nullptr, 10000);
+    }
+
+    const double qppr = QPPR.get();
+    const double qpps_vmax = Vmax.get() * qppr / (2 * M_PI * Rw.get()); // Maximum linear velocity interm of qpps
+    const double qp_amax = Amax.get() * DEG_TO_RAD * qppr / (2 * M_PI); // Maximum angle interm of quadratur pulses (qp)
+    const double dz = DZ.get() * qp_amax / 100.0;                       // Dead zone around target position
```

Comment:
Is double really necessary? We want to avoid it as much as possible, as the most used boards (STM32F4) don't have hardware support for it.